### PR TITLE
`azurerm_container_group` - Support resource limit for cpu/mem/gpu

### PR DESCRIPTION
**Pull request by magodo**
This PR adds support for `{cpu|memory|gpu}_limit` for `azurerm_container_group`. The difference between (e.g.) `cpu_limit` and `cpu` should be the same semantic as is for [k8s](https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-resource-requests-and-limits). Another difference is that the *limit* set properties are optional and updatable.

## Test

```shell
❯ TF_ACC=1 go test -timeout=60m -v -run=TestAccContainerGroup_linuxComplete ./internal/services/containers
=== RUN   TestAccContainerGroup_linuxComplete
=== PAUSE TestAccContainerGroup_linuxComplete
=== CONT  TestAccContainerGroup_linuxComplete
--- PASS: TestAccContainerGroup_linuxComplete (1499.90s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    1499.908s
```

### CODE REVIEW

File: internal/services/containers/container_group_resource.go [hunk 1]
```diff
@@ -291,6 +291,41 @@ func resourceContainerGroup() *pluginsdk.Resource {
...
 
+						"cpu_limit": {
+							Type:     pluginsdk.TypeFloat,
+							Optional: true,
+						},
+
+						"memory_limit": {
+							Type:     pluginsdk.TypeFloat,
+							Optional: true,
+						},
```
Comment (katbyte) [inline internal/services/containers/container_group_resource.go:302]: we should be validaing these? also waht sort of limit? we should include the units in the property names

File: internal/services/containers/container_group_resource.go [hunk 2]
```diff
@@ -291,6 +291,41 @@ func resourceContainerGroup() *pluginsdk.Resource {
...
+						"gpu_limit": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"count": {
+										Type:     pluginsdk.TypeInt,
+										Optional: true,
+									},
```
Comment (katbyte) [inline internal/services/containers/container_group_resource.go:314]: validation?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/containers/container_group_resource.go (hunk 1 above)
Comment (magodo) [reply internal/services/containers/container_group_resource.go]: The naming of these properties follow the convention as the non limit ones: https://github.com/hashicorp/terraform-provider-azurerm/blob/c268eea425c73e286206ed816cec4f2a8ad8dc08/internal/services/containers/container_group_resource.go#L249-L292

The unit of them are described in the document.

**Comment by magodo** [comment]
@katbyte Thank you for the review! I've added the validations for them, but still prefer keeping the naming of them as is to be consistent with the non-limit properties. While the unit of these properties are mentioned in the document already.

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/containers/container_group_resource.go [hunk 3]
```diff
@@ -291,6 +291,44 @@ func resourceContainerGroup() *pluginsdk.Resource {
 							},
 						},
 
+						"cpu_limit": {
```
[the remark this answers is not in the document]
Comment (magodo) [reply internal/services/containers/container_group_resource.go:294]: Yes, e.g. 0.5 is a valid value.

File: internal/services/containers/container_group_resource.go [hunk 4]
```diff
@@ -291,6 +291,44 @@ func resourceContainerGroup() *pluginsdk.Resource {
...
+							Type:         pluginsdk.TypeFloat,
+							Optional:     true,
+							ValidateFunc: validation.FloatAtLeast(0.0),
```
[the remark this answers is not in the document]
Comment (magodo) [reply internal/services/containers/container_group_resource.go:297]: Per the document https://docs.microsoft.com/en-us/azure/container-instances/container-instances-quotas#service-quotas-and-limits:

> Availability of compute, memory, and storage resources for Azure Container Instances varies by region and operating system.
> ...
> To request a limit increase, create an [Azure Support request](https://portal.azure.com/#blade/Microsoft_Azure_Support/HelpAndSupportBlade/newsupportrequest)

So there isn't a proper number to be set as the upper limit here. E.g. I noticed my subscription can at most apply for 400 cores at west europe.

File: internal/services/containers/container_group_resource.go [hunk 5]
```diff
@@ -291,6 +291,44 @@ func resourceContainerGroup() *pluginsdk.Resource {
...
+						"gpu_limit": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"count": {
+										Type:         pluginsdk.TypeInt,
+										Optional:     true,
+										ValidateFunc: validation.IntAtLeast(0),
```
[the remark this answers is not in the document]
Comment (magodo) [reply internal/services/containers/container_group_resource.go:316]: Same here

**Pull request merged**